lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
Yup. I don't like it. In the case of ruby parser, there was a case that ran fine in jruby (probably a bug, if not just a semantic difference) but raised in MRI. It was a RegexpError and not in the list. If RubyParser raises an exception, it is almost invariably because of something like this. Logging it to stderr helped me _find_ the bug, not hide it. For whatever reason, the way things currently worked the whole thing broke down and none of stderr was printed. This fixes it so I can actually get info and fix bugs further down.